ipld/read.go

```diff
@@ -247,3 +248,40 @@ func GetLeaf(ctx context.Context, dag ipld.NodeGetter, root cid.Cid, leaf, total
 	// recursively walk down through selected children
 	return GetLeaf(ctx, dag, root, leaf, total)
 }
+
+// GetLeavesByNamespace returns all the shares from the given DataAvailabilityHeader root
+// with the given namespace.ID.
+func GetLeavesByNamespace(
+	ctx context.Context,
+	dag ipld.NodeGetter,
+	root cid.Cid,
+	nID namespace.ID) (out []ipld.Node, err error) {
```

Comment:
nit: I would add a `,` after `namespace.ID` and move the rest of the method signature in a separate line.